tracer/src/Datadog.Trace/AppSec/InstrumentationGateway.cs

```diff
@@ -100,8 +103,11 @@ private void RaiseEvent(HttpContext context, Span relatedSpan, Func<IDictionary<
             {
                 var eventData = getEventData();
                 var transport = new HttpTransport(context);
-                LogAddressIfDebugEnabled(eventData);
-                eventHandler.Invoke(this, new InstrumentationGatewaySecurityEventArgs(eventData, transport, relatedSpan, eraseExistingAddress));
+                if (!transport.Blocked)
+                {
+                    LogAddressIfDebugEnabled(eventData);
+                    eventHandler?.Invoke(this, new InstrumentationGatewaySecurityEventArgs(eventData, transport, relatedSpan, eraseExistingAddress));
+                }
```

Comment:
Same as above comment.

Reply:
This skips the debug log too though, is that correct?

Reply:
I would say yes, we only really want to know what the input data is if it's going to be analyzed by the waf.